applications/Chat/evaluate/metric.py

```diff
@@ -0,0 +1,177 @@
+import statistics
+
+import jieba
+from bert_score import score
+from rouge_chinese import Rouge as Rouge_cn
+from nltk.translate.bleu_score import sentence_bleu
+from sklearn.metrics import precision_score, recall_score, f1_score
+
+
+def bleu_score(preds: list, targets: list) -> dict:
+    """Calculate BLEU Score Metric
+
+    The calculation includes BLEU-1 for unigram, BLEU-2 for bigram, 
+    BLEU-3 for trigram and BLEU-4 for 4-gram. Unigram evaluates the
+    accuracy in word level, other n-gram evaluate the fluency in 
+    sentence level.
+    """
+    bleu_scores = {"bleu1": 0, "bleu2": 0, "bleu3": 0, "bleu4": 0}
+    cumulative_bleu = [0]*4
+    weights = [
+        (1./1., 0., 0., 0.),
+        (1./2., 1./2., 0., 0.),
+        (1./3., 1./3., 1./3., 0.),
+        (1./4., 1./4., 1./4., 1./4.)
+     ]
+
+    for pred, target in zip(preds, targets):
+        pred_list = (' '.join(jieba.cut(pred))).split()
+        target_list = [(' '.join(jieba.cut(target))).split()]
+
+        bleu = sentence_bleu(target_list, pred_list, weights = weights)
+        cumulative_bleu = [a+b for a, b in zip(cumulative_bleu, bleu)]
+
+    bleu_scores["bleu1"] = cumulative_bleu[0]/len(preds)
```

Comment:
Can you use a for-loop to simplify this?

Reply:
By the way, do you install pre-commit? It seems that the format is not correct.

Reply:
```
for i in range(len(cumulative_bleu)):
        bleu_scores[f"bleu{i+1}"] = cumulative_bleu[i] / len(preds) 
```